api/v1/__init__.py

```diff
@@ -0,0 +1,182 @@
+import yaml
+from fastapi import APIRouter, BackgroundTasks, HTTPException
+from git.repo import Repo
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+from sqlmodel import Session
+
+from apps.backport.datasync.datasync import upload_gzip_dict
+from etl import config, paths
+from etl import grapher_model as gm
+from etl.command import main as etl_main
+from etl.db import get_engine
+from etl.files import yaml_dump
+from etl.helpers import read_json_schema
+from etl.metadata_export import merge_or_create_yaml, reorder_fields
+from etl.paths import SCHEMAS_DIR
+
+from .. import utils
+from .schemas import Indicator, UpdateIndicatorRequest
+
+log = structlog.get_logger()
+
+
+engine = get_engine()
+
+DATASET_SCHEMA = read_json_schema(path=SCHEMAS_DIR / "dataset-schema.json")
+
+
+v1 = APIRouter()
+
+
+@v1.get("/api/v1/health")
+def health() -> dict:
+    return {"status": "ok"}
+
+
+@v1.put("/api/v1/indicators")
+def update_indicator(update_request: UpdateIndicatorRequest, background_tasks: BackgroundTasks):
+    _validate_data_api_url(update_request.dataApiUrl)
+
+    # update YAML file
+    with Session(engine) as session:
+        db_indicator = gm.Variable.load_from_catalog_path(session, update_request.catalogPath)
+
+    if db_indicator.catalogPath is None:
+        raise HTTPException(403, "Only indicators from the ETL can be edited. Contact us if you really need this.")
+
+    override_yml_path, table_name, indicator_short_name = _parse_catalog_path(str(db_indicator.catalogPath))
+
+    # convert incoming indicator updates into dictionary
+    meta_dict = _indicator_metadata_dict(update_request.indicator, table_name, indicator_short_name)
+
+    yaml_str = merge_or_create_yaml(
+        meta_dict,
+        override_yml_path,
+    )
+
+    # reorder YAML and dump it back again in nice format
+    yaml_str = yaml_dump(reorder_fields(yaml.safe_load(yaml_str)))
+    assert yaml_str
+
+    # validate YAML against our schema
+    try:
+        validate(instance=yaml.safe_load(yaml_str), schema=DATASET_SCHEMA)
+    except ValidationError as e:
+        raise HTTPException(status_code=422, detail=e.message)
+
+    # NOTE: the order of all updating "tracks" is random, we can rethink it if we find
+    # ourselves running into race conditions.
+    if not update_request.dryRun:
+        with open(override_yml_path, "w") as f:
+            f.write(yaml_str)
+
+    # try to commit and push before overwriting file in R2
+    if config.ETL_API_COMMIT:
+        _commit_and_push(override_yml_path, ":robot: Metadata update by Admin")
+
+    if not update_request.dryRun:
+        _update_indicator_in_r2(db_indicator, update_request.indicator)
+
+    # trigger ETL in the background (this is usually fast, but for some datasets this could take a while)
+    if update_request.triggerETL:
+        background_tasks.add_task(_trigger_etl, indicator_short_name, db_indicator, update_request.dryRun)
+
+    return {"yaml": yaml_str}
```

Comment:
For later / nit: this function would be more readable if is was 5 lines that read like pseudocode, calling out to smaller functions that do the work.

Reply:
I tried to break it down to functions. It's not exactly 5 lines, but better than before.